Hyperion.Tests/ISerializableTests.cs

```diff
@@ -7,9 +7,177 @@
+            {
+                info.AddValue("FirstName", FirstName);
+                info.AddValue("LastName", LastName);
+            }
+        }
+
+        public sealed class PersonWithoutConstructor : ISerializable
+        {
+            public PersonWithoutConstructor(string firstName)
+            {
+                FirstName = firstName;
+            }
+
+            public string FirstName { get; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext streamingContext)
+            {
+                info.AddValue("FirstName", FirstName);
+            }
+        }
+
+        public sealed class PersonWithIDeserializationCallback : ISerializable, IDeserializationCallback
+        {
+            public PersonWithIDeserializationCallback() {}
+
+            public PersonWithIDeserializationCallback(SerializationInfo info, StreamingContext context)
+            {
+                FirstName = (string)info.GetValue("FirstName", typeof(string));
+            }
+
+            public string FirstName { get; set; }
+
+            public void GetObjectData(SerializationInfo info, StreamingContext streamingContext)
+            {
+                info.AddValue("FirstName", FirstName);
+            }
+
+            public void OnDeserialization(object sender)
+            {
+                throw new Exception("OnDeserialization");
+            }
+        }
+        #endregion
+
+        [Fact]
+        public void CanIgnoreFieldsWithNonSerializedAttribute()
+        {
+            var expected = new NonSerializedTest() {
+                Field1 = 235,
+                Field2 = "non serialized text"
+            };
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<NonSerializedTest>();
+            Assert.Equal(expected.Field1, actual.Field1);
+            Assert.Null(actual.Field2);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void SkipNonSerializedAttributeIfNoSerilizedAttribute()
+        {
+            var expected = new NonSerializedWithoutSerializableTest() {
+                Field1 = 235,
+                Field2 = "non serialized text"
+            };
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<NonSerializedWithoutSerializableTest>();
+            Assert.Equal(expected.Field1, actual.Field1);
+            Assert.Equal(expected.Field2, actual.Field2);
+        }
+
+        [Fact]
+        public void CanSerializeClassesWithISerializable()
+        {
+            var expected = new Person("Scott", "Hanselman");
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<Person>();
+            Assert.Equal(expected.FirstName, actual.FirstName);
+            Assert.Equal(expected.LastName, actual.LastName);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void ShouldNotThrowIfNoConstructorWithSerializationInfo()
```

Comment:
I'm not sure, what should we do, if we can't find a proper constructor for ISerializable